shell/dialog/ScaleMachineDownDialog.vue

```diff
@@ -13,31 +13,41 @@ export default {
     }
   },
 
+  created() {
+    this.isRke2 = this.resources[0].cluster?.isRke2;
+  },
+
   async fetch() {
-    await Promise.all([
-      this.$store.dispatch('management/findAll', { type: CAPI.MACHINE_DEPLOYMENT }),
-      this.$store.dispatch('management/findAll', { type: CAPI.MACHINE })
-    ]);
+    if (this.isRke2) {
+      await Promise.all([
+        this.$store.dispatch('management/findAll', { type: CAPI.MACHINE_DEPLOYMENT }),
+        this.$store.dispatch('management/findAll', { type: CAPI.MACHINE })
+      ]);
+    } else {
+      await Promise.all([
+        this.$store.dispatch('management/findAll', { type: MANAGEMENT.NODE_POOL }),
+        this.$store.dispatch('management/findAll', { type: MANAGEMENT.NODE })
+      ]);
+    }
   },
 
   data() {
-    const allToDelete = Array.isArray(this.resources) ? this.resources : [this.resources];
-    const cluster = allToDelete[0].cluster;
+    const cluster = this.isRke2 ? this.resources[0].cluster : this.resources[0].provisioningCluster;
```

Comment:
@richard-cox the safety check for the resources to be an array is removed because it was a duplicate, we already have it on `PromptModal.vue`
https://github.com/rancher/dashboard/blob/91d756af80975b448f0a3d3424392c9686ee0362/shell/components/PromptModal.vue#L22